nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--config ${cfg.confFile}"
+    "--work-dir ${cfg.workDir}"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ optional (cfg.logFile != null) "--logfile ${cfg.logFile}"
+  ++ optional (cfg.pidFile != null) "--pidfile ${cfg.pidFile}"
+  ++ optional cfg.verbose "--verbose");
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    confFile = mkOption {
+      default = "/etc/AdGuardHome.yaml";
+      type = path;
+      description = ''
+        Path to the config file.
+      '';
+    };
```

Comment:
Please follow https://github.com/NixOS/rfcs/pull/42.

Reply:
I think having a non-mutable config file will not work well for this service. Management via the webui does not work and the config file is migrated automatically between schema versions on some application updates.

Maybe it would be better, as it is mutable, to store the config file in the work dir by default? Anyone who really wants to try for a stateless config can then override `confFile` and use `environment.etc`.

Reply:
The application also fails the initial set-up if the config file is not writable.